fetchai/ledger/parser/etch.grammar

```diff
@@ -18,41 +18,41 @@ instruction: "use" NAME [use_shard] ";"                         -> use_global
 
 instantiation: "var" NAME "=" expression ";"
 
-declaration: "var" NAME ":" TYPE ";"
+declaration: "var" NAME ":" type ";"
 
 range: (expression) | (expression ":" expression) | (expression ":" expression ":" expression)
 
 expression: NUMBER
-          | dot_expansion
           | name
+          | dot_expansion
           | STRING_LITERAL
           | PRE_UNARY expression
           | expression POST_UNARY
           | expression BINARY_OP expression
           | function_call
-          | TYPE input_block
+          | type input_block
 
 function_call: expression input_block
 
-dot_expansion: name ("." name)+
+dot_expansion: expression ("." expression)+
 
 bool_expression: expression
                | PRE_UNARY bool_expression
                | bool_expression COMPARISON bool_expression
                | bool_expression BOOLEAN_OP bool_expression
 
 annotation: "@"ANNOTATION function
-ANNOTATION: "init"|"action"|"query"|"problem"|"objective"|"work"|"clear"
+ANNOTATION: "initTest"|"init"|"action"|"query"|"problem"|"objective"|"work"|"clear"|"test"
 
-function: "function" NAME parameter_block (":" TYPE)? code_block? "endfunction"
+function: "function" NAME parameter_block (":" type)? code_block? "endfunction"
 
 parameter_block: "(" (typed_name ","?)* ")"
 
-typed_name: NAME ":" TYPE
+typed_name: NAME ":" type
 
 input_block: "(" (expression ",")* expression? ")"
 
-template_block: "<" TYPE ">"
+template_block: "<" type ">"
```

Comment:
this doesn't look right - shouldn't a template block be a comma-separated list of types?

Reply:
Correct - this isn't actually used though, it duplicates the correct definition below. Removed it.